client/src/components/LibraryNavigator/LibraryModel.ts

```diff
@@ -0,0 +1,197 @@
+  private sessionId: string;
+
+  public async connect(): Promise<void> {
+    const session = getSession();
+    let computeSession = null;
+
+    await window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: "Connecting to SAS session...",
+      },
+      async () => {
+        computeSession = await session.setup();
+      }
+    );
+
+    this.sessionId = computeSession && computeSession.sessionId;
+    this.dataAccessApi = DataAccessApi(getApiConfig());
+  }
+
+  public async setup(): Promise<void> {
+    if (this.sessionId && this.dataAccessApi) {
+      return;
+    }
+
+    await this.connect();
+  }
+
+  public async loadViewData(item: LibraryItem): Promise<TableData> {
+    await this.setup();
+    const response = await this.retryOnFail(
+      async () =>
+        await this.dataAccessApi.getRows({
+          sessionId: this.sessionId,
+          libref: item.library,
+          tableName: item.name,
+          includeColumnNames: true,
+        })
+    );
+
+    return {
+      headers: response.data.items[0],
+      rows: response.data.items.slice(1),
+    };
+  }
+
+  public async getTable(item: LibraryItem) {
+    await this.setup();
+    const response = await this.retryOnFail(
+      async () =>
+        await this.dataAccessApi.getTable({
+          sessionId: this.sessionId,
+          libref: item.library,
+          tableName: item.name,
+        })
+    );
+
+    return response.data;
+  }
+
+  public async deleteTable(item: LibraryItem) {
+    await this.setup();
+    try {
+      await this.retryOnFail(
+        async () =>
+          await this.dataAccessApi.deleteTable({
+            sessionId: this.sessionId,
+            libref: item.library,
+            tableName: item.name,
+          })
+      );
+    } catch (error) {
+      throw new Error(
+        sprintf(Messages.TableDeletionError, { tableName: item.uid })
+      );
+    }
+  }
+
+  public async getChildren(item?: LibraryItem): Promise<LibraryItem[]> {
+    if (!item) {
+      return await this.getLibraries();
+    }
+
+    return await this.getTables(item);
+  }
+
+  private async getLibraries(): Promise<LibraryItem[]> {
+    await this.setup();
+    const options = {
+      headers: { Accept: "application/vnd.sas.collection+json" },
+    };
+
+    const {
+      data: { items },
+    } = await this.retryOnFail(
+      async () =>
+        await this.dataAccessApi.getLibraries(
+          { sessionId: this.sessionId },
+          options
+        )
+    );
+
+    const libraryItems: LibraryItem[] = await Promise.all(
+      items.map(async (item: LibraryItem): Promise<LibraryItem> => {
+        const { data } = await this.retryOnFail(
+          async () =>
+            await this.dataAccessApi.getLibrarySummary({
+              sessionId: this.sessionId,
+              libref: item.id,
+            })
+        );
+
+        return { ...item, readOnly: data.readOnly };
+      })
+    );
+
+    libraryItems.push({
```

Comment:
This looks quite odd. Is there a reason for this?

Reply:
Sure. `this.dataAccessApi.getLibraries` doesn't return a work library. This just manually adds one

Reply:
getLibraries returns all assigned libraries, including WORK. If you are not seeing WORK returned, we should figure out why

Reply:
It turns out that on the test server, there were 11 libraries and the default page size is 10. Updating the code to handle paging will fix this.